Tasks/AzureResourceManagerTemplateDeploymentV3/operations/Utils.ts

```diff
@@ -407,6 +422,62 @@ class Utils {
     private static escapeBlockCharacters(str: string): string {
         return str.replace(/[\[]/g, '$&[]');
     }
+
+    private static async getFilePathForLinkedArtifact(filePath: string): Promise<string> {
+        var filePathExtension: string = filePath.split('.').pop();
+        if(filePathExtension === 'bicep'){
+            let azcliversion = await this.getAzureCliVersion()
+            if(parseFloat(azcliversion)){
+                if(this.isBicepAvailable(azcliversion)){
+                    await this.execBicepBuild(filePath)
+                    filePath = filePath.replace('.bicep', '.json')
+                    this.cleanupFileList.push(filePath)
+                }else{
+                    throw new Error(tl.loc("IncompatibleAzureCLIVersion"));
+                }
+            }else{
+                throw new Error(tl.loc("AzureCLINotFound"));
+            }
+        }
+        return filePath
+    }
+
+    private static async getAzureCliVersion(): Promise<string> {
+        let azcliversion: string = "" ;
+        try {
+            const { stdout, stderr } = await cpExec('az version');
+            if (!stderr) {
+                azcliversion = JSON.parse(stdout)["azure-cli"]
+            } else {
+                throw stderr
+            }
+        } catch (err) {
+            throw new Error(tl.loc("FailedToFetchAzureCLIVersion", err));
+        }
+        return azcliversion
+    }
+
+    private static async execBicepBuild(filePath): Promise<void> {
+        try {
+            const { stdout, stderr } = await cpExec(`az bicep build --file ${filePath}`);
+            if (!stderr) {
+            } else {
+                throw stderr
+            }
+        } catch (err) {
+            throw new Error(tl.loc("BicepBuildFailed", err));
+        }
+    }
+
+    private static isBicepAvailable(azcliversion): Boolean{
+        let majorVersion = azcliversion.split('.')[0]
+        let minorVersion = azcliversion.split('.')[1]
+        // Support Bicep was introduced in az-cli 2.20.0
+        if(majorVersion >= 2 && minorVersion >= 20){
```

Comment:
I think this should be:
`(majorVersion == 2 && minorVersion >= 20) || majorVersion > 2)`

Otherwise you'll block e.g. version `3.10`

Reply:
Ah thats right 😅